pkg/scheduler/service/cleaner.go

```diff
@@ -67,154 +69,194 @@ func (c *cleaner) purgeReconciliations(transition *ClusterStatusTransition, conf
 
 //Purges reconciliations using rules from: https://github.com/kyma-incubator/reconciler/issues/668
 func (c *cleaner) purgeReconciliationsNew(transition *ClusterStatusTransition, config *CleanerConfig) {
-	now := time.Now()
 
-	latestReconciliations, err := c.getLatestReconciliations(config.keepLatestEntitiesCount(), transition)
+	clusters, err := transition.inventory.GetAll()
```

Comment:
I think this can be a very expensive operation if we get all clusters, is there any way we can instead use a query here that already takes into account the age through config.maxEntitiesAgeDays() and the count and status criteria? Maybe I overestimate this, but I think that getting all clusters can have thousands of records through which you iterate both `deleteRecordsByAge` and `deleteRecordsByCountAndStatus`, even if they end up as NoOp. Maybe a delete driven by the database is more efficient here. Let me know what you think :) I see that you are using these methods maybe also for readability and configurability but I think the delete could also be written as one single query